tests/unit/commands/migration/test_migration.py

```diff
@@ -71,6 +75,112 @@ def test_do_list_return_no_migration_message_when_there_is_no_migration(self, mo
+        mock_check_credAccount.return_value = uforge.CredAccountVSphere()
+        mock_create_Migration.return_value = uforge.migration()
+        m = migration.Migration()
+        m.api = Api("url", username="username", password="password", headers=None,
+                    disable_ssl_certificate_validation=False, timeout=constants.HTTP_TIMEOUT)
+        m.login = "login"
+        m.password = "password"
+
+        mock_deploy_and_launch_agent.return_value(1)
+
+        # when
+        m.do_launch("--file config.json")
+
+        # then
+        mock_out.assert_called_with("file: config.json is validated", "OK")
+        self.assertEquals(mock_download.call_count, 1)
+
+    @patch("hammr.commands.migration.Migration.download_migration_binary")
+    @patch("ussclicore.utils.printer.out")
+    def test_do_launch_display_message_error_when_file_is_not_filled(self, mock_out, mock_download):
+        # given
+        m = migration.Migration()
+
+        # when
+        m.do_launch("--file")
+
+        # then
+        mock_out.assert_called_with("ERROR: In Arguments: argument --file: expected one argument", "ERROR")
+        self.assertEquals(mock_download.call_count, 0)
+
+    @patch("hammr.commands.migration.Migration.download_migration_binary")
+    @patch("ussclicore.utils.generics_utils.get_file")
+    @patch("ussclicore.utils.printer.out")
+    def test_do_launch_not_launch_download_migration_binary_when_file_is_not_found(self, mock_out, mock_get_file,
+                                                                                   mock_download):
+        # given
+        mock_get_file.return_value = None
+        m = migration.Migration()
+
+        # when
+        m.do_launch("--file config.json")
+
+        # then
+        mock_out.assert_called_with("No such file or directory: config.json", "ERROR")
+        self.assertEquals(mock_download.call_count, 0)
+
+    @patch("hammr.commands.migration.Migration.download_migration_binary")
+    @patch("ussclicore.utils.generics_utils.get_file")
+    @patch("ussclicore.utils.printer.out")
+    @patch("hammr.utils.hammr_utils.load_data")
+    def test_do_launch_not_launch_download_migration_binary_when_configuration_file_do_not_contain_migration_parameter(
+            self,
+            mock_load_data,
+            mock_out,
+            mock_get_file,
+            mock_download):
+        # given
+        mock_get_file.return_value = "a file"
+        mock_load_data.return_value = "something wrong"
+        m = migration.Migration()
+
+        # when
+        m.do_launch("--file config.json")
+
+        # then
+        mock_out.assert_called_with("no migration section found", "ERROR")
+        self.assertEquals(mock_download.call_count, 0)
+
+    def test_createMigration_create_a_migration_with_all_stages(self):
+        # given
+        m = migration.Migration()
+        data = self.retrieve_jsonFile("tests/integration/data/testMigrate.json")
```

Comment:
close to IT, maybe we could build the json

Reply:
done